demos/shadow/shadow_demo_main/shadow_demo_main.c

```diff
@@ -0,0 +1,644 @@
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief This example uses the MQTT library of the AWS IoT Device SDK for
+ * Embedded C. This is the prototype of the callback function defined by
+ * that library. It will be invoked whenever the MQTT library receives an
+ * incoming message.
+ *
+ * @param[in] pMqttContext MQTT context pointer.
+ * @param[in] pPacketInfo Packet Info pointer for the incoming packet.
+ * @param[in] pDeserializedInfo Deserialized information from the incoming packet.
+ */
+static void eventCallback( MQTTContext_t * pMqttContext,
+                           MQTTPacketInfo_t * pPacketInfo,
+                           MQTTDeserializedInfo_t * pDeserializedInfo );
+
+/**
+ * @brief Process payload from /update/delta topic.
+ *
+ * This handler examines the version number and the powerOn state. If powerOn
+ * state has changed, it sets a flag for the main function to take further actions.
+ *
+ * @param[in] pPublishInfo Deserialized publish info pointer for the incoming
+ * packet.
+ */
+static void updateDeltaHandler( MQTTPublishInfo_t * pPublishInfo );
+
+/**
+ * @brief Process payload from /update/accepted topic.
+ *
+ * This handler examines the accepted message that carries the same clientToken
+ * as sent before.
+ *
+ * @param[in] pPublishInfo Deserialized publish info pointer for the incoming
+ * packet.
+ */
+static void updateAcceptedHandler( MQTTPublishInfo_t * pPublishInfo );
+
+/*-----------------------------------------------------------*/
+
+static void updateDeltaHandler( MQTTPublishInfo_t * pPublishInfo )
+{
+    static uint32_t currentVersion = 0; /* Remember the latestVersion # we've ever received */
+    uint32_t version = 0U;
+    uint32_t newState = 0U;
+    char * outValue = NULL;
+    uint32_t outValueLength = 0U;
+
+    assert( pPublishInfo != NULL );
+    assert( pPublishInfo->pPayload != NULL );
+
+    LogInfo( ( "/update/delta json payload:%s.\n\n", pPublishInfo->pPayload ) );
+    /* Make sure the payload is json document. */
+    if ( JSONSuccess == JSON_Validate( pPublishInfo->pPayload,
+                                       pPublishInfo->payloadLength ) )
+    {
+        /* In this demo, we discard the incoming message
+         * if the version number is not newer than the latest
+         * that we've received before. Your application may use a
+         * different approach. 
+         */
+
+        /* Then we start to get the version value by JSON keyword "version".
+         * The payload will look similar to this:
+         * {
+         *      "version": 12,
+         *      "timestamp": 1595437367,
+         *      "state": {
+         *          "powerOn": 1
+         *      },
+         *      "metadata": {
+         *          "powerOn": {
+         *          "timestamp": 1595437367
+         *          }
+         *      },
+         *      "clientToken": "388062"
+         *  }
+         */
+        JSONStatus_t result = JSONSuccess;
```

Comment:
declare at the beginning of the function with other variables.

Reply:
This [commit](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1104/commits/49b40ee1665d75b8ed186c326788b6200655e313) should fix the comment.
please let me know if you have other questions.